zipline/_protocol.pyx

```diff
@@ -462,16 +478,27 @@ cdef class BarData:
             })
 
     cdef bool _can_trade_for_asset(self, asset, dt, adjusted_dt, data_portal):
-        session_label = normalize_date(dt) # FIXME
+        cdef object session_label
+        cdef object dt_to_use_for_exchange_check,
+        cdef object dt_to_use_for_last_price_check
+
+        session_label = self._trading_calendar.minute_to_session_label(dt)
+
         if not asset.is_alive_for_session(session_label):
             # asset isn't alive
             return False
 
-        # FIXME temporarily commenting out while we sort out some downstream
-        # dependencies
-        # if not asset.is_exchange_open(dt):
-        #     # exchange isn't open
-        #     return False
+        if not self._daily_mode:
+            # Find the next market minute for this calendar, and check if this
+            # asset's exchange is open at that minute.
+            if self._trading_calendar.is_open_on_minute(dt):
+                dt_to_use_for_exchange_check = dt
+            else:
+                dt_to_use_for_exchange_check = \
+                    self._trading_calendar.next_open(dt)
+
+            if not asset.is_exchange_open(dt_to_use_for_exchange_check):
+                return False
```

Comment:
upon rereading this, just realized that the `is_open_on_minute` check, upon returning True, should skip the rest of this block.  I'll fix.